controller_manager/test/test_controllers_chaining_with_controller_manager.cpp

```diff
@@ -610,6 +679,271 @@ TEST_P(
+
+  // Test Case 1: Trying to activate a preceding controller when following controller is not activated
+  // --> return error (If STRICT); Preceding controller is still inactive.
+
+  // There is different error and timeout behavior depending on strictness
+  auto const getExpectedBehaviorVals = [&]
+  {
+    if (
+      test_param.strictness ==
+      controller_manager_msgs::srv::SwitchController::Request::STRICT)  // STRICT
+    {
+      return std::make_tuple(
+        controller_interface::return_type::ERROR, std::future_status::ready,
+        lifecycle_msgs::msg::State::PRIMARY_STATE_ACTIVE);
+    }
+    else if (
+      test_param.strictness ==
+      controller_manager_msgs::srv::SwitchController::Request::BEST_EFFORT)  // BEST EFFORT
+    {
+      return std::make_tuple(
+        controller_interface::return_type::OK, std::future_status::timeout,
+        lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE);
+    }
+    else
+    {
+      std::runtime_error("test_param was not defined!");
+    }
+  };
+  auto const expected = getExpectedBehaviorVals();
+
+  // Attempt to activate preceding controller (diff-drive controller) with no check
+  ActivateController(DIFF_DRIVE_CONTROLLER, std::get<0>(expected), std::future_status::ready);
+
+  // Check if the controller activated (Should not be activated)
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, diff_drive_controller->get_state().id());
+
+  // Test Case 2: Try to activate a preceding controller the same time when trying to deactivate a following controller (using switch_controller function)
+  // --> return error; preceding controller is not activated, BUT following controller IS deactivated
+
+  // Activate and check the following controllers (pid_left_wheel_controller) (pid_right_wheel_controller)
+  ActivateAndCheckController(
+    pid_left_wheel_controller, PID_LEFT_WHEEL, PID_LEFT_WHEEL_CLAIMED_INTERFACES, 1u);
+  ActivateAndCheckController(
+    pid_right_wheel_controller, PID_RIGHT_WHEEL, PID_RIGHT_WHEEL_CLAIMED_INTERFACES, 1u);
+
+  // Attempt to activate a preceding controller (diff-drive controller) while trying to deactivate a following controller
+  switch_test_controllers(
+    {DIFF_DRIVE_CONTROLLER}, {PID_LEFT_WHEEL, PID_RIGHT_WHEEL}, test_param.strictness,
+    std::get<1>(expected), std::get<0>(expected));
+
+  // Preceding controller should stay deactivated and following controller should be deactivated (if BEST_EFFORT)
+  EXPECT_EQ(std::get<2>(expected), pid_left_wheel_controller->get_state().id());
+  EXPECT_EQ(std::get<2>(expected), pid_right_wheel_controller->get_state().id());
+  ASSERT_EQ(
+    lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, diff_drive_controller->get_state().id());
+}
+
+TEST_P(
+  TestControllerChainingWithControllerManager,
+  test_chained_controllers_activation_switching_error_handling)
+{
+  // Test Case 3: In terms of current implementation. Example: Need two diff drive controllers, one should be deactivated,
```

Comment:
I am not sure here, even from the description, what we are trying to test. Can you please help me? This should never send an error, is this correct?